secretflow/ml/nn/fl/fl_model.py

```diff
@@ -619,18 +623,24 @@ def save_model(self, model_path: Union[str, Dict[PYU, str]], is_test=False):
         res = []
         for device, worker in self._workers.items():
             assert device in model_path, f'Should provide a path for device {device}.'
-            if not os.path.exists(model_path[device]):
-                os.makedirs(model_path[device])
+            device_model_path, device_model_name = model_path[device].rsplit("/", 1)
+
             if is_test:
-                model_path_test = os.path.join(
-                    model_path[device], device.__str__().strip("_")
+                device_model_path = os.path.join(
+                    device_model_path, device.__str__().strip("_")
                 )
-                res.append(worker.save_model(model_path_test))
-            else:
-                res.append(worker.save_model(model_path[device]))
+            if not os.path.exists(device_model_path):
```

Comment:
`if not os.path.exists(device_model_path)` creates dirs on driver node, this behavior is expected to done on device node, so worker creates dir locally is necessary but not here.